features/system-report/src/main/java/org/opennms/systemreport/event/CsvUtils.java

```diff
@@ -0,0 +1,196 @@
+    public static final String USER_LOGINS_CSV_FILE_PATH = String.join("", OPENNMS_HOME, "/etc/user_logins.csv");
+    private static final String  USER_NAME_CSV_HEADERS = "User Name";
+    private static final String  LOGIN_TIME_CSV_HEADERS = "Login Time";
+    private static final String  USER_LOGINS_CSV_HEADERS = String.join(",", USER_NAME_CSV_HEADERS, LOGIN_TIME_CSV_HEADERS);
+    private static final Integer USER_LOGIN_THRESHOLD_DAYS= 60;
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+
+    public static Resource readCsvHeaders(String filePath) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (Reader reader = new FileReader(filePath, StandardCharsets.UTF_8);
+             CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT.withFirstRecordAsHeader())) {
+
+            String[] headers = csvParser.getHeaderMap().keySet().toArray(new String[0]);
+            try (Writer writer = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
+                 CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT)) {
+
+                csvPrinter.printRecord((Object[]) headers);
+                csvPrinter.flush();
+            }
+        }
+
+        return new ByteArrayResource(outputStream.toByteArray());
+    }
+
+    public static Resource readCsvData(String filePath, boolean skipHeaders) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (Reader reader = new FileReader(filePath, StandardCharsets.UTF_8);
+             CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT)) {
+            try (Writer writer = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
+                 CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT)) {
+                boolean isFirstRow = true;
+                for (CSVRecord record : csvParser) {
+                    if (skipHeaders && isFirstRow) {
+                        isFirstRow = false;
+                        continue;
+                    }
+                    csvPrinter.printRecord(record);
+                }
+                csvPrinter.flush();
+            }
+        }
+        return new ByteArrayResource(outputStream.toByteArray());
+
+    }
+
+    public static void writeDataToFile(Resource csvHeaders, Resource csvData, String filePath) throws IOException {
+
+        File csvFile = new File(filePath);
+        boolean isNewFile = !csvFile.exists() && csvFile.createNewFile();
+        try (FileOutputStream fileOutputStream = new FileOutputStream(csvFile, true)) {
+            writeDataToStream(csvHeaders, csvData, fileOutputStream, isNewFile);
+        }
+    }
+
+    public static void writeDataToStream(Resource csvHeaders, Resource csvData, OutputStream outputStream,boolean writeHeaders) throws IOException {
```

Comment:
Please format this file properly.
```suggestion
    public static void writeDataToStream(Resource csvHeaders, Resource csvData, OutputStream outputStream, boolean writeHeaders) throws IOException {
```